source/common/network/listen_socket_impl.cc

```diff
@@ -37,6 +38,22 @@ TcpListenSocket::TcpListenSocket(const Address::InstanceConstSharedPtr& address,
   int rc = setsockopt(fd_, SOL_SOCKET, SO_REUSEADDR, &on, sizeof(on));
   RELEASE_ASSERT(rc != -1);
 
+#if defined SOL_IP && defined IP_TRANSPARENT
+  // Set the SOL_IP/IP_TRANSPARENT socket option to support running with iptables TPROXY target
+  // instead of REDIRECT to get both the original source and destination addresses of the
+  // redirected socket.
+  on = 1;
+  rc = setsockopt(fd_, SOL_IP, IP_TRANSPARENT, &on, sizeof(on));
```

Comment:
I don't think this should be set unconditionally; this should only be done based on some configuration.

Reply:
+1

Reply:
Done.